app.rb

```diff
@@ -0,0 +1,116 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'classroom'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each { |person| puts "#{person.class}: Name: #{person.name}, Id: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    case person_type
+    when '1'
+      student = create_student
+      @people << student
+      puts 'Student created successfully'
+    when '2'
+      teacher = create_teacher
+      @people << teacher
+      puts 'Teacher created successfully'
+    else
+      puts 'Invalid option'
+    end
+  end
+
```

Comment:
- [ ] Great job working on the functionality so far but when I try to create a new person I face this error wrong number of arguments (given 4, expected 3) so kindly check on this error and try to solve the problem to let the user be able to create a person whether a student or a teacher ⭐🔥💯🙏🏻

- When I try to create a student  ⬇
![image](https://github.com/ClaudiaRojasSoto/OOP_School_Library/assets/58553711/7afbe570-8d29-49d4-ba8a-067b429d2fed)

- When I try to create a teacher⬇
![image](https://github.com/ClaudiaRojasSoto/OOP_School_Library/assets/58553711/84969189-5eaa-44a8-89dc-5c4670b04ce3)

Reply:
Thank you @Reem-lab. I solved the problems and now the app is 100% functional.